### ISSUE TITLE
Coc still open preview with completeopt-=preview 

### ISSUE THREAD

**Issue by sandangel** [neoclide/coc.nvim#216]
**Result from CocInfo**
Run `:CocInfo` command and paste the content below.
```sh
## versions

vim version: NVIM v0.3.2-dev
node version: v11.3.0
coc.nvim version: 0.0.35
term: screen-256color
platform: darwin

## Error messages

## Output channel: tsserver
[Info  - 1:30:08 PM] Forking TSServer
PATH: /usr/local/bin:/Users/sand/.cargo/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/sand/Works/Go/bin:/Users/sand/Library/Android/sdk/tools:/Users/sand/Library/Android/sdk/tools/bin:/Users/sand/Library/Android/sdk/platform-tools:/Users/sand/Library/Android/sdk/emulator:/Users/sand/.yarn/bin:/Users/sand/Works/Flutter/flutter/bin:/Users/sand/.antigen/bundles/robbyrussell/oh-my-zsh/lib:/Users/sand/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/git:/Users/sand/.antigen/bundles/zsh-users/zsh-syntax-highlighting:/Users/sand/.antigen/bundles/zsh-users/zsh-completions:/Users/sand/.antigen/bundles/zsh-users/zsh-history-substring-search:/Users/sand/.antigen/bundles/zsh-users/zsh-autosuggestions:/Users/sand/.antigen/bundles/lukechilds/zsh-nvm:/Users/sand/.pub-cache/bin:/Users/sand/.fzf/bin:/Users/sand/.yarn/bin:/Users/sand/.pub-cache/bin 
[Info  - 1:30:08 PM] Started TSServer
{
  "path": "/Users/sand/Downloads/verifone-challenge/node_modules/typescript/lib",
  "_api": {
    "versionString": "3.1.6",
    "version": "3.1.6"
  }
}

## Output channel: tslint
[Info  - 1:30:09 PM] TSLint library loaded from: /Users/sand/Downloads/verifone-challenge/node_modules/tslint/lib/index.js
Warning: The 'deprecation' rule requires type information.
Warning: The 'no-use-before-declare' rule requires type information.

```
**Describe the bug**

Coc Open preview even when I set completeopt-=preview

**To Reproduce**
set completeopt-=preview

`:CocConfig`, write `coc` and pick the first item.

the preview window show

**Screenshots**

![dec-01-2018 13-48-46](https://user-images.githubusercontent.com/22189661/49324394-e95a8a00-f56f-11e8-9f52-8652ce316a32.gif)

Another issue you can see in the gif is coc does not select the first item although I set `noselect` to false in the `CocConfig`

**Comment by oblitum** [comment]
Isn't there another thing resetting your `completeopt`? I can't reproduce this, even replicating the gif.

**Comment by oblitum** [comment]
Check `:verbose set completeopt` for example.

**Comment by sandangel** [comment]
@oblitum yeah, it's is `completeopt=menu,preview,menuone,noinsert`

Let's me try disable ALE.

**Comment by oblitum** [comment]
Execute `:verbose set completeopt` once you enter the `coc-settings.json`, before typing in there. It should show the last time it was set and which script did it.

**Comment by sandangel** [comment]
@oblitum it only shows the string I wrote above.

**Comment by oblitum** [comment]
It's with preview from `.vimrc`.

**Comment by oblitum** [comment]
Post a link to your `.vimrc`. Despite what you state about the conf, NeoVim reports it has been last set with `preview`.

**Comment by sandangel** [comment]
@oblitum thanks, here is [the link](https://github.com/sandangel/dotfiles)

**Comment by chemzqm** [comment]
It's special for json file, this is made so you can always get the description of settings, maybe I should restrict it to coc-settings.json.

**Comment by sandangel** [comment]
@chemzqm hi I don't think it fixed. I just run update and it recompile from source.

![dec-01-2018 14-31-35](https://user-images.githubusercontent.com/22189661/49324676-01cda300-f576-11e8-917d-d30002211300.gif)

**Comment by oblitum** [comment]
The fix on da753ca looks like still activating it for `coc-settings.json`, but personally I'd prefer it to be left as I set in my `.vimrc`.

**Comment by sandangel** [comment]
@oblitum yeah I think coc should respect the setting in vimrc or it might be difficult to debug when conflict with other plugin

**Comment by sandangel** [comment]
@oblitum I notice that the preview window only show with those complete candidates end with [S], might be a `ultisnips` problem?

**Comment by oblitum** [comment]
A bit off-topic, but seems there's a typo in the whole file, `activted` instead of `activted`,  or `Activted`  instead of `Activted`. It seems there's no issue with that, since the typo is consistent.

**Comment by chemzqm** [comment]
>  I think coc should respect the setting in vimrc

It can't, `noinsert,menuone` must be in `completopt`, otherwise completion won't work.

> candidates end with [S], 

`[S]` is snippets from extensions, not ultisnipt.

**Comment by chemzqm** [comment]
>  I just run update and it recompile from source.

Try rebuild again and restart your vim.

**Comment by oblitum** [comment]
> > I think coc should respect the setting in vimrc
> 
> It can't, `noinsert,menuone` must be in `completopt`, otherwise completion won't work.
> 

But...,  my `completopt` is just `menu` and it seems to be working fine.

**Comment by chemzqm** [comment]
>  my completopt is just menu and it seems to be working fine.

coc change your completopt and restore it back.

**Comment by oblitum** [comment]
ok.

**Comment by sandangel** [comment]
@chemzqm so how could I prevent show preview window with the snippet?

I just completely removed `coc.nvim` and reinstall it
```sh
~/.dotfiles master
❯ cd ~/.vim/plugged/

~/.vim/plugged
❯ rm -rf coc.nvim

~/.vim/plugged
❯ nvim +PlugInstall

~/.vim/plugged 15s
❯
```
the preview window is still showing

**Comment by chemzqm** [comment]
Try `:verbose set completeopt` just after open your file, if `preview` is there, find out where it's added.

**Comment by oblitum** [comment]
@sandangel I've reproduced!, Indeed... :(

**Comment by oblitum** [comment]
I mean, I've reproduced inside `coc-settings.json`, though I wasn't expecting since I removed preview in the `.vimrc`.

**Comment by sandangel** [comment]
@chemzqm ok I fixed it with explicitly `set completeopt=menu,menuone,noinsert`
